app/src/client/wasm.rs

```diff
@@ -28,19 +28,57 @@ pub fn initialize(world: &mut World) -> anyhow::Result<()> {
 
     std::thread::spawn(move || {
         let stream = ambient_audio::AudioStream::new().unwrap();
+        let mut sound_info = std::collections::HashMap::new();
         while let Ok(message) = rx.recv() {
             match message {
-                AudioMessage::Track(t, looping, amp) => {
-                    match looping {
-                        true => {
-                            let sound = stream.mixer().play(t.decode().repeat().gain(amp.clamp(0.0, 1.0)));
-                            sound.wait();
+                AudioMessage::Track(t, looping, amp, url, uid) => {
+                    let gain = Arc::new(Mutex::new(amp));
+                    let gain_clone = gain.clone();
+
+                    let sound = match looping {
+                        true => stream.mixer().play(t.decode().repeat().gain(gain_clone)),
+                        false => stream.mixer().play(t.decode().gain(gain_clone)),
+                    };
+                    sound.wait();
+                    sound_info.insert(uid, (url, looping, gain, sound.id));
```

Comment:
I'd suggest using a struct instead of a tuple here

Reply:
fixed